api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -911,32 +950,75 @@ type AzureNodePoolPlatform struct {
+	// DiskEncryptionSetID should also exist in a resource group under the same subscription id and the same location
+	// listed in the Hosted Cluster, hcluster.Spec.Platform.Azure.Location.
+	//
 	// +optional
 	DiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
-	// EnableEphemeralOSDisk enables ephemeral OS disk
+
+	// EnableEphemeralOSDisk is a flag when set to true, will enable ephemeral OS disk.
+	//
 	// +optional
 	EnableEphemeralOSDisk bool `json:"enableEphemeralOSDisk,omitempty"`
+
 	// SubnetID is the subnet ID of an existing subnet where the nodes in the nodepool will be created. This can be a
 	// different subnet than the one listed in the HostedCluster, hcluster.Spec.Platform.Azure.SubnetID, but must exist
 	// in the same hcluster.Spec.Platform.Azure.VnetID and must exist under the same subscription ID,
 	// hcluster.Spec.Platform.Azure.SubscriptionID.
 	//
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SubnetID is immutable"
 	// +kubebuilder:validation:Required
-	// +immutable
-	// +required
 	SubnetID string `json:"subnetID"`
+
 	// Diagnostics specifies the diagnostics settings for a virtual machine.
-	// If not specified then Boot diagnostics will be disabled.
+	// If not specified, then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
 }
 
+// MarketplaceImage specifies the information needed to create an Azure VM from an Azure Marketplace image. This struct
+// replicates the same fields found in CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go.
+type MarketplaceImage struct {
+	// Publisher is the name of the organization that created the image
+	//
+	// +kubebuilder:validation:MinLength=1
+	Publisher string `json:"publisher"`
+
+	// Offer specifies the name of a group of related images created by the publisher.
+	//
+	// +kubebuilder:validation:MinLength=1
+	Offer string `json:"offer"`
+
+	// SKU specifies an instance of an offer, such as a major release of a distribution.
+	// For example, 18.04-LTS, 2019-Datacenter
+	//
+	// +kubebuilder:validation:MinLength=1
+	SKU string `json:"sku"`
+
+	// Version specifies the version of an image sku. The allowed formats are Major.Minor.Build or 'latest'. Major,
+	// Minor, and Build are decimal numbers. Specify 'latest' to use the latest version of an image available at
+	// deployment time. Even if you use 'latest', the VM image will not automatically update after deploy time even if a
+	// new version becomes available.
+	//
+	// +kubebuilder:validation:MinLength=1
+	Version string `json:"version"`
+
+	// ThirdPartyImage indicates the image is published by a third party publisher and a Plan will be generated for it.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	ThirdPartyImage bool `json:"thirdPartyImage"`
```

Comment:
Prefer not to use bools, but enums with meaningful string values, they evolve better

Reply:
Ack, I don't have a strong preference either way. @enxebre do you have a preference here? 

I implemented it as a bool since CAPZ has this as a bool value so I thought it would be easier to just copy the value over from our API when creating the Azure machine template.

Reply:
do we even need this field?

Reply:
I'll drop it. I talked with @bennerv and it doesn't seem like we need it - https://redhat-internal.slack.com/archives/C05Q232L29E/p1720706367758479?thread_ts=1718196807.708239&cid=C05Q232L29E.